tests/lint/check_asf_header.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -e
+#!/usr/bin/env -S bash -e
```

Comment:
While you're cleaning these up could you move all the `bash -e` usages into a `set -e` after the apache header everywhere? That way we can have a consistent bash invocation across all scripts

```suggestion
#!/usr/bin/env bash
```

Reply:
Updated. Actually it won't build without this change. The coreutil in the CI docker image might be old and doesn't have the `-S` arg for `env`.

Reply:
Some scripts already have `set -euxo pipefail` so I didn't add `set -e` to them